language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AssignmentStatementNodeContext.java

```diff
@@ -41,13 +57,57 @@ public AssignmentStatementNodeContext() {
     public List<LSCompletionItem> getCompletions(LSContext context, AssignmentStatementNode node)
             throws LSCompletionException {
         List<LSCompletionItem> completionItems = new ArrayList<>();
-        completionItems.addAll(this.actionKWCompletions(context));
-        completionItems.addAll(this.expressionCompletions(context));
+        if (this.onQualifiedNameIdentifier(context, node.expression())) {
+            /*
+            Captures the following cases
+            (1) [module:]TypeName c = module:<cursor>
+            (2) [module:]TypeName c = module:a<cursor>
+             */
+            QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) node.expression();
+            Predicate<Scope.ScopeEntry> filter = scopeEntry -> {
+                BSymbol symbol = scopeEntry.symbol;
+                return symbol instanceof BVarSymbol && (symbol.flags & Flags.PUBLIC) == Flags.PUBLIC;
+            };
+            List<Scope.ScopeEntry> moduleContent = QNameReferenceUtil.getModuleContent(context, qNameRef, filter);
+            completionItems.addAll(this.getCompletionItemList(moduleContent, context));
+        } else {
+            /*
+            Captures the following cases
+            (1) [module:]TypeName c = <cursor>
+            (2) [module:]TypeName c = a<cursor>
+             */
+            completionItems.addAll(this.actionKWCompletions(context));
+            completionItems.addAll(this.expressionCompletions(context));
+            completionItems.addAll(this.getNewExprCompletionItems(context, node));
+            completionItems.add(new SnippetCompletionItem(context, Snippet.KW_IS.get()));
+        }
         return completionItems;
     }
 
     @Override
     public boolean onPreValidation(AssignmentStatementNode node) {
         return !node.equalsToken().isMissing();
     }
+
+    private List<LSCompletionItem> getNewExprCompletionItems(LSContext context, AssignmentStatementNode node) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        ArrayList<Scope.ScopeEntry> visibleSymbols = new ArrayList<>(context.get(CommonKeys.VISIBLE_SYMBOLS_KEY));
+        Optional<BObjectTypeSymbol> objectType;
```

Comment:
this local variable seems to be redundant. can we add merge  `ifPresent` logic in https://github.com/ballerina-platform/ballerina-lang/pull/25214/files#diff-3fef1a396b3d9462c79c38bf09b51ca3R108 into https://github.com/ballerina-platform/ballerina-lang/pull/25214/files#diff-3fef1a396b3d9462c79c38bf09b51ca3R103. And else block can be safely removed. WDYT?

Reply:
+1
Sending this in next commit.